src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -1184,6 +1212,14 @@ void TableDockWidget::deleteSelectedItems()
     if (selectedItems.isEmpty())
         return;
 
+    // checks if the selected item count is same as the no. of 
+    // groups in the table.
+    if (topLevelItemsCount == topLevelGroupCount()) {
+        auto continueDeletion = deleteAllgroupsWarning();
+        if (!continueDeletion) {
+          return;
+        }
+    }
```

Comment:
Issue #853 was about warning the user when the select "Delete All Groups" from the context menu as well. This check can still be useful. If the `topLevelItemsCount == topLevelGroupCount()` is true then you simply call `TableDockWidget::deleteAll` method, instead of the complex logic that follows which is totally unnecessary since the user has asked to indiscriminately wipe out each item the table.